config.ru

```diff
@@ -1,14 +1,24 @@
 # frozen_string_literal: true
 
 require './config/environment'
+require 'securerandom'
+require 'rack/protection'
 
 if ActiveRecord::Base.connection.migration_context.needs_migration?
   raise 'Migrations are pending. Run `rake db:migrate` to resolve the issue.'
 end
 
 use Rack::MethodOverride
-
-require 'rack/protection'
-use Rack::Protection, use: %i[content_security_policy], script_src: "'self'", img_src: "'self'"
+use Rack::Session::Cookie,
+    domain: ->(env) { Rack::Request.new(env).host },
+    path: '/',
+    expire_after: 3600 * 24,
+    secret: ENV.fetch('SESSION_SECRET') { SecureRandom.hex(64) }
+use Rack::Protection,
+    use: %i[content_security_policy authenticity_token],
+    permitted_origins: ENV.fetch('URL', nil),
+    script_src: "'self'",
+    script_src_attr: "'self'",
+    script_src_elem: "'self'"
```

Comment:
Why do we need that? According to Mozilla script-src-elem and script-src-attr are already covered by script-src (https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy/script-src)